core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -461,7 +461,7 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
           appListener.appAttemptId,
           appListener.startTime.getOrElse(-1L),
           appListener.endTime.getOrElse(-1L),
-          fileStatus.getModificationTime(),
+          clock.getTimeMillis(),
```

Comment:
As an attempt at keeping this value a little more a reflection of reality, I think it would be better to keep `fileStatus.getModificationTime()` if the application is complete. Otherwise, for complete applications, this will reflect the time the history server started in a lot of cases.